scylla/src/transport/session.rs

```diff
@@ -2237,12 +2226,16 @@ impl Drop for RequestSpan {
     }
 }
 
-fn partition_key_displayer<'pk>(
-    prepared_metadata: &'pk PreparedMetadata,
-    partition_key: &'pk [u8],
-) -> impl Display + 'pk {
+fn partition_key_displayer<'ps, 'res>(
+    mut pk_values_iter: impl Iterator<Item = (&'ps [u8], &'ps ColumnSpec)> + 'res + Clone,
+) -> impl Display + 'res {
     CommaSeparatedDisplayer(
-        PartitionKeyDecoder::new(prepared_metadata, partition_key).map(|c| match c {
+        std::iter::from_fn(move || {
+            pk_values_iter
+                .next()
+                .map(|(mut cell, spec)| deser_cql_value(&spec.typ, &mut cell))
+        })
```

Comment:
Why is `std::iter::from_fn` needed? Isn't just `pk_values_iter.map(|(mut cell, spec)| deser_cql_value(&spec.typ, &mut cell))` sufficient?

Reply:
I did overengineer this, I admit, due to my fascination with `from_fn` once I've recently discovered it. Enexpectedly, however, I run into strange lifetime issues once I change it to `map`. Could you yourself try to refactor it as you suggest?

Reply:
I tried, but I gave up quickly.